From 19911dfb290261e858e911c1fe91e428a79f7be1 Mon Sep 17 00:00:00 2001
From: Liwei Song <liwei.song@windriver.com>
Date: Tue, 2 Jun 2015 10:20:38 +0800
Subject: [PATCH 349/349] random: fix the compiled warning in rt kernel with
 seed definition

This patch is to fix the warning below:
drivers/char/random.c:815:17: warning: unused variable 'seed'
[-Wunused-variable]
  unsigned long  seed;
                 ^
To fix this, in rt kernel we do not use this variable, use RT config
to control it

Signed-off-by: Liwei Song <liwei.song@windriver.com>
---
 drivers/char/random.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/char/random.c b/drivers/char/random.c
index dc681d6..4e227ee 100644
--- a/drivers/char/random.c
+++ b/drivers/char/random.c
@@ -812,7 +812,9 @@ void add_interrupt_randomness(int irq, int irq_flags, __u64 ip)
 	struct fast_pool	*fast_pool = &__get_cpu_var(irq_randomness);
 	unsigned long		now = jiffies;
 	__u32			input[4], cycles = get_cycles();
+#ifndef CONFIG_PREEMPT_RT_FULL
 	unsigned long		seed;
+#endif
 	int			credit = 0;
 
 	input[0] = cycles ^ jiffies;
-- 
1.7.5.4

